### ISSUE TITLE
Add scm decorations to recently used editors

### ISSUE THREAD

**Issue by usernamehw** [microsoft/vscode#90301]
For the recently used menu (which is bound to <kbd>Ctrl</kbd> + <kbd>Tab</kbd>) I would like to see git decoration status (modified/untracked/...) just like for **OPEN EDITORS** View.

![Screenshot (142)](https://user-images.githubusercontent.com/9638156/74091650-a9bff400-4aca-11ea-82ba-0218fe7a50e5.png)

**Comment by bpasero** [comment]
I like the idea, but have to check if that is possible with quick pick today.

**Comment by bpasero** [comment]
I briefly looked into this and it seems this is currently not possible with quick input because it is using `IconLabel` and not `ResourceLabel` (which has the support for decorations). Leaving up to @chrmarti to think about it.

**Comment by bpasero** [comment]
Oh, this is actually non trivial because resource labels live in workbench and require services such as the decorations service. That probably reduces chances to get this in, given we just moved quick input down to base.

**Comment by bpasero** [comment]
Oh, this is actually easier than I had thought because decorations only need to add some classes to the target item and that API already exists, e.g. here:

https://github.com/microsoft/vscode/blob/f01d9ab212535aef8a221b31de0e53193ad7adcf/src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts#L879-L879

Visually adds a lot more noise:

![image](https://user-images.githubusercontent.com/900690/133963819-3d4b6056-f528-493c-81ae-5b4664de04a5.png)

But we can probably play with the decoration options to tweak that. To begin with we can maybe only show the badge and nothing else.

**Comment by TylerLeonhardt** [comment]
Made a ton of implementations of this and I'm not sure I'm pleased with any of them:

simply adding the classes in
![image](https://user-images.githubusercontent.com/2644648/174199985-ab20ceec-9f26-46ad-ac01-bb38c57757de.png)
no label color
![image](https://user-images.githubusercontent.com/2644648/174199997-aa9bab20-63f8-469c-aec6-249faccbb300.png)

moving the dirty to the left
![image](https://user-images.githubusercontent.com/2644648/174200030-27cd29d4-f5dc-4b04-8a11-6d1c3e898d55.png)
no label color
![image](https://user-images.githubusercontent.com/2644648/174200044-15f54d97-16bf-44b2-9eb5-053595d1ff66.png)

Another place for the dirty
![image](https://user-images.githubusercontent.com/2644648/174200062-6bbf954b-1fdb-4d6a-bcd1-738ce38313b6.png)
Another place for the dirty
![image](https://user-images.githubusercontent.com/2644648/174200070-2039873f-5743-413c-87ad-0261f44a9d00.png)

Couple of other ideas from chat that are a bit beyond my CSS skills atm
* @Tyriar: "Could the dot be to the left of the decorations on the right side?"
* @roblourens bringing up how he's configured tabs today:
![image](https://user-images.githubusercontent.com/2644648/174200243-b7c4e5a1-2e0d-4a1d-af06-9e38cacde94a.png)
* @lramos15: "could you replace the dots with 1, M, Or U and then have those be colorized so it's just one column?"

**Comment by bpasero** [comment]
Can we not just put the decorations right next to the label instead of right aligning? Similar to tabs where the decoration is right next to the label.

**Comment by TylerLeonhardt** [comment]
Yeah I personally would like the decorations to be a single line for easy scanning. I wonder if I could move the badge to the left next to the icon... but to me it does feel more natural on the right since dirty status is also listed there.

**Comment by stuartcrobinson** [comment]
thank you for working on this!

my vote:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/12928716/221295149-052be8ab-cf14-453a-9c38-a938ac7b6de2.png">

(i also removed "recently opened" cos i'm not sure what it adds, feels like clutter to me personally)

**Comment by Tyriar** [comment]
My preference:

- Show color on the name, even though there might be contrast problems, as just the indicator on the right could easily be missed
  - I wonder if we should have a terminal minimum contrast ratio-like feature for cases like this, instead of introducing new colors for each possible background color, we just shift luminance of the foreground color as needed to meet 1:4.5 cc @daviddossett 
- Show both SCM and problems on the right, but keep space low
- Show dirty indicator to the left of the file as it's important info

![image](https://github.com/microsoft/vscode/assets/2193314/fd6c28e2-0eaa-43f3-b53a-8fefd1c1f4c1)

**Comment by Tyriar** [comment]
Also it's not clear why the decoration would not align with the `x`:

![image](https://github.com/microsoft/vscode/assets/2193314/1aba18fa-a715-4572-a1f2-bc6c8a6b6c0d)

**Comment by Tyriar** [comment]
More thoughts:

- Make it more like a table on the right?
- Do we need to show recently opened at all? Isn't there only a single section?

![image](https://github.com/microsoft/vscode/assets/2193314/98f365a8-c38f-4037-93a8-b6abef2a280a)

**Comment by TylerLeonhardt** [comment]
> Make it more like a table on the right?

I think this is ideal, but the implementation would be challenging. The dot is a button (which is an `x` on hover for the other items), while the decoration is well the decoration... not a button. Your mockup puts the decoration in between the buttons (the `[|]` in another button)

> Do we need to show recently opened at all? Isn't there only a single section?

It's the only section when there's no filter. With a filter, there is another section.
